File: pinyougou_manager_web/src/main/webapp/js/controller/brandController.js
//自定义控制层
app.controller("brandController",function ($scope,$controller,brandService) {

    //控制器继承代码
    //参数一,继承的父控制器  参数二共享$scope对象
    $controller("baseController",{$scope:$scope})




    $scope.findAll=function () {
        brandService.findAll().success(
            function (response) {
                $scope.list=response;
            })
    }


    //分页查询方法
    $scope.search=function (pageNum,pageSize) {
        brandService.findPage(pageNum,pageSize).success(
            function(response){
                $scope.list = response.rows;
                $scope.paginationConf.totalItems=response.total;
            }
        );
    }

    $scope.findOne=function (id) {
        brandService.findOne(id).success(
            function (response) {
                $scope.entity=response;
            }
        )
    }

    $scope.save=function () {
        if($scope.entity.id != null){
            brandService.update($scope.entity).success(
                function (response) {
                    if (response.success){
                        $scope.reloadList();
                    }else {
                        alert(response.message);
                    }
                }
            )
        }else {
            brandService.add($scope.entity).success(
                function (response) {
                    if (response.success){
                        $scope.reloadList();
                    }else {
                        alert(response.message);
                    }
                }
            )
        }
    }




    $scope.delete=function () {
        if(confirm("你确认要删除吗?")){
            brandService.delete($scope.selectIds).success(
                function (response) {
                    if(response.success){
                        $scope.reloadList();
                    }else {
                        alert(response.message)
                    }
                }
            )
        }
    }
});

File: pinyougou_manager_web/src/main/java/com/pinyougou/manager/controller/BrandController.java
package com.pinyougou.manager.controller;
import com.alibaba.dubbo.config.annotation.Reference;
import com.pinyougou.pojo.TbBrand;
import com.pinyougou.sellergoods.service.BrandService;
import entity.PageResult;
import entity.Result;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

import java.util.List;
import java.util.Map;

/**
 * Created by ZJE on 2018/12/3
 */


@RestController
@RequestMapping("/brand")
public class BrandController {

    @Reference
    private BrandService brandService;

    //查询所有展示所有商品
    @RequestMapping("/findAll")
    public List<TbBrand> findAll(){

        return brandService.findAll();
    }

    //分页查询商品
    @RequestMapping("/findPage")
    public PageResult findPage(Integer pageNum,Integer pageSize){
        return brandService.findPage(pageNum,pageSize);
    }

    //新增商品
    @RequestMapping("/add")
    public Result add(@RequestBody TbBrand brand){
        try{
            brandService.add(brand);
            return new Result(true,"添加成功!");
        }catch (Exception e){
            e.printStackTrace();
            return new Result(false,"添加失败");
        }
    }

    //修改商品
    @RequestMapping("/update")
    public Result update(@RequestBody TbBrand brand){
        try{
            brandService.update(brand);
            return new Result(true,"修改成功!");
        }catch (Exception e){
            e.printStackTrace();
            return new Result(false,"修改失败");
        }
    }

    //根据id查询商品
    @RequestMapping("/findOne")
    public TbBrand findOne(Long id){
        return brandService.findOne(id);
    }

    //根据id删除商品
    @RequestMapping("/delete")
    public Result delete(Long[] ids){
        try{
            brandService.delete(ids);
            return new Result(true,"修改成功!");
        }catch (Exception e){
            e.printStackTrace();
            return new Result(false,"修改失败");
        }
    }

    @RequestMapping("/selectBrandList")
    public List<Map> selectBrandList(){
        return brandService.selectBrandList();
    }






}
